### PR TITLE
feat: adds flag to leave k8s cluster intact when restore fails & configure timeout

### PR DESCRIPTION
**Pull request by evadnoob**
Adds flag to ignore snapshot restore failures

### CODE REVIEW

File: cmd/devenv/provision/provision.go [hunk 1]
```diff
@@ -708,24 +716,29 @@ func (o *Options) Run(ctx context.Context) error { //nolint:funlen,gocyclo // Wh
...
+				if err2 != nil {
+					o.log.WithError(err).Error("failed to remove intermediate environment")
+					return err2
+				}
 
-			return errors.Wrap(err, "failed to provision from snapshot")
+				return errors.Wrap(err, "failed to provision from snapshot")
+			} else {
+				o.log.Warn("")
+				o.log.WithError(err).Warn("failed to provision from snapshot, destroy devenv is disabled")
```
Comment (jaredallard) [inline cmd/devenv/provision/provision.go:740]:
```suggestion
				o.log.WithError(err).Warn("failed to provision from snapshot, leaving devenv intact (--destroy-on-failure is false)")
```

File: cmd/devenv/provision/provision.go [hunk 2]
```diff
@@ -708,24 +716,29 @@ func (o *Options) Run(ctx context.Context) error { //nolint:funlen,gocyclo // Wh
...
+			}
```
Comment (jaredallard) [inline cmd/devenv/provision/provision.go:741]: I think we should just return here. I don't think we want to do anything else if we failed.

Comment (jaredallard) [reply cmd/devenv/provision/provision.go:741]: (this is for `destroy-on-failure` being false)

Comment (evadnoob) [reply cmd/devenv/provision/provision.go:741]: ok, sounds good, was debating that too

**Review by jaredallard — APPROVED** [review]

File: cmd/devenv/provision/provision.go [hunk 3]
```diff
@@ -111,7 +121,7 @@ func (o *Options) MarshalLog(addField func(key string, v interface{})) {
 	addField("devenv.provision.snapshot_channel", o.SnapshotChannel)
 	addField("devenv.provision.base", o.Base)
 	addField("devenv.provision.use_devspace", o.UseDevspace)
-
+	addField("devenv.provision.destroy_on_failures", o.DestroyOnFailure)
```
Comment (evadnoob) [inline cmd/devenv/provision/provision.go:124]:
```suggestion
	addField("devenv.provision.destroy_on_failure", o.DestroyOnFailure)
```

**Pull request merged**